lib/heapsort.js

```diff
@@ -1,8 +1,24 @@
+const { MinHeap } = require('./minheap');
 // This method uses a heap to sort an array.
-// Time Complexity:  ?
-// Space Complexity: ?
+// Time Complexity:  O(n log n) 
+// Space Complexity: O(n + log n) because of the length of the array and the heap storage-> O(n)
 function heapsort(list) {
-  throw new Error('Method not implemented yet...');
+  let minHeap = new MinHeap();
+
+  for (let num of list) {
+    minHeap.add(num);
+  }
+
+  let i = 0;
+  let currentMin;
+
+  while (!minHeap.isEmpty()) {
+    currentMin = minHeap.remove();
+    list[i] = currentMin;
+    i++;
+  }
+  
+  return list;
 };
```

Comment:
Note the since this isn't a fully in-place solution (the MinHeap has a O(n) internal store), we don't necessarily need to modify the passed in list. The tests are written to check the return value, so we could unpack the heap into a new result list to avoid mutating the input.

```py
    const result = [];
    while (!minHeap.isEmpty()) {
        result.push(minHeap.remove())
    }

    return result;
```